int64 case_num
---
bool success